src/tailer/__init__.py

```diff
@@ -159,6 +160,8 @@ def follow(self, delay=1.0):
         trailing = True
 
         while 1:
+            if not os.path.exists(os.path.realpath(self.file.name)): #Check if the file still exists
```

Comment:
`self.file` may be a stream instead of a file with a name. So `self.file.name` might result in an attribute error. There are a couple of ways I can think of to handle this:

1. If `getattr(self.file, "name")`, then assume it's a file and check for existence. However, this could be an expensive call on a long file since it will cause an extra disk read for every single line.
2. Do a non-blocking readline and only `yield` something was read. This might be able to be done with [select](https://docs.python.org/3/library/select.html). I'm not certain how it will react if the file disappears. This will be a little more complex and might involve some more refactoring, but it should result in the best experience.

I'm inclined to choose option 2. Since you opened this pull request 5 years ago, you probably don't care anymore 😅. I'll see about fixing this. Thanks for the submission!